collector/benchmarks/README.md

```diff
@@ -246,3 +246,51 @@ Rust code being written today.
 - Compare the benchmarking time of the two versions.
 - When adding the new version, for `perf-config.json` and the `N-*.patch`
   files, use the corresponding files for the old version as a starting point.
+
+# Benchmark update policy
+
+rustc-perf is a "living benchmark suite" that is regularly changed. Some
+benchmarks in rustc-perf are verbatim copies of third-party crates. We
+periodically do a mass update of these benchmarks.
+
+Benefits of this approach:
+- We ensure we are measuring compilation of the crates most people are using.
+  This is most relevant for popular crates.
+- We get coverage of newer language features.
+
+Costs of this approach:
+- It takes time and effort.
+- We lose some data continuity.
+  - But the stable set of benchmarks used for the dashboard are not affected,
+    and they provide the greatest continuity.
+- If the code hasn't changed much, it won't have much effect.
+
+To balance these requires choosing a good refresh period. A period of three
```

Comment:
Do we want to be in the habit of refreshing all at once like we just did? Would it not be better to offset upgrading benchmarks? This would minimize overall disruption and limit the amount of work at any one given time.

Reply:
Upgrading all of the benchmarks at once is not fun, but at least it's a concentrated effort and the timeframe is well defined. I'm a bit worried that if the upgrade was "continuous", the stability of the perf. suite would degrade, if e.g. every month of a few months new benchmarks would appear or disappear. It could surprise the users of perf.rlo if the benchmarks are changing too often.

Reply:
If we don't have a mass update policy, then we need an alternative incremental update policy. That feels much more nebulous to me, and likely to lead to updates not happening.